packages/netlify-cms-core/src/reducers/entries.ts

```diff
@@ -123,11 +192,74 @@ const entries = (state = Map({ entities: Map(), pages: Map() }), action: Entries
       });
     }
 
+    case SORT_ENTRIES_REQUEST: {
+      const payload = action.payload as EntriesSortRequestPayload;
+      const { collection, key, direction } = payload;
+      const newState = state.withMutations(map => {
+        const sort = OrderedMap({ [key]: Map({ key, direction }) });
+        map.setIn(['sort', collection], sort);
```

Comment:
This is where I dropped the multi-sort logic to minimize the impact - all other supporting logic is still in place.